### PR TITLE
profile metadata workaround

### PR DESCRIPTION
**Pull request by arlimus**
@chris-rock let's review this use-case

Background: If we run compliance profiles through kitchen-inspec, all of the compliance metadata gets lost.

### CODE REVIEW

File: lib/kitchen/verifier/inspec.rb [hunk 1]
```diff
@@ -36,12 +36,24 @@ class Inspec < Kitchen::Verifier::Base
 
       default_config :inspec_tests, []
 
+      def add_target(runner, target, opts)
```
Comment (chris-rock) [inline lib/kitchen/verifier/inspec.rb]: @vjeffrey @arlimus I am not sure why we are re-implementing https://github.com/chef/inspec/blob/1f5185638669136d962e0550aad3281c0d4d7630/lib/inspec/runner.rb#L49 here

**Comment by chris-rock** [comment]
@vjeffrey Could you elaborate the use-case a little bit more? Please correct me if I am wrong, but wouldn't it be better to extend the inspec runner to be able to return the information? I propose we store the profiles in InSpec runner: https://github.com/chef/inspec/blob/1f5185638669136d962e0550aad3281c0d4d7630/lib/inspec/runner.rb#L50

File: lib/kitchen/verifier/inspec.rb (hunk 1 above)
Comment (srenatus) [inline lib/kitchen/verifier/inspec.rb]: I guess this https://github.com/chef/kitchen-inspec/pull/74/files#diff-873cf55477008bb3a2a80b3fa94e7a06R42 is the reason, the "profile metadata workaround". @chris-rock are you proposing a different workaround?

**Comment by arlimus** [comment]
@chris-rock the use-case was to run the hardening-ssh and similar test profiles, that were structured into `test/integration/*...` where it mainly had all of the tests collected without any `inspec` folder, but was completely missing an `inspec.yml` for metadata. We need metadata, since we want to display it in the CI/CD pipeline. i.e. full use-case: Take a mixed profile and make it work in CI/CD with full metadata.

This may not be necessary after all. Can you provide an alternative in case this use-case is already possible?

**Comment by vjeffrey** [comment]
I was just looking into this really quickly: the functionality we had tried to introduce in this branch isn't working anymore. I tested by cloning webapp-server (from dcc) and getting the profile in the right place and running bundle exec ktichen test. On both master and this branch, the profile information showed up as `"summary_line":"10 examples, 0 failures","profiles":[{"supports":[]}]}`.  So, we don't end up with the profile info...(see here: https://delivery.chef.co/e/chef/#/organizations/audit/projects/webapp-server/changes/443414d2-efea-489f-a2cc-1419b3e03d29/status/acceptance). We're expecting to get the title, version, and summary in the delivery ui (https://github.com/chef/delivery/blob/64c8bffc1fb26a7284e3b8d118b3d7789bc214ed/web/src/components/log/log_compliance.html, https://github.com/chef/delivery/blob/master/web/src/components/log/log_compliance.js#L86)...help?

**Comment by chris-rock** [comment]
This feature is covered with native inspec. We use `runner.add_target` which loads profiles properly and loads the metadata in: https://github.com/chef/inspec/blob/master/lib/inspec/runner.rb#L72-L88

Challenge was, that combination of json formatter and profiles caused errors in inspec. To retrieve the profile metadata, InSpec Profile implementation instantiates a mock runner and loads a complete profile. This crashed in cases, where the profile was using custom resources.

The use case of this PR should be covered in https://github.com/chef/inspec/pull/704

**Comment by chris-rock** [comment]
Please reopen, if this is still required

**Pull request closed without merging**